Python: Azure AI Search Memory doesn't allow TokenCredential.

**Describe the bug**
[azure_cognitive_search_memory_store.py](https://github.com/microsoft/semantic-kernel/blob/b250109328d8b284d24682420945df5cd3057bd4/python/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py#L83) module seems to allow the use of a TokenCredential such as from DefaultAzureCredenial() or one of its variants. Line 83 of the code explicitly checks for the API key being present, thus it never allows the TokenCredential to be used to create the SearchIndexClient in the [utils.py](https://github.com/microsoft/semantic-kernel/blob/b250109328d8b284d24682420945df5cd3057bd4/python/semantic_kernel/connectors/memory/azure_cognitive_search/utils.py) at [line 79](https://github.com/microsoft/semantic-kernel/blob/b250109328d8b284d24682420945df5cd3057bd4/python/semantic_kernel/connectors/memory/azure_cognitive_search/utils.py#L79).

**To Reproduce**
Steps to reproduce the behavior:
1. In a Jupyter Notebook try to create AzureCognitiveSearchMemoryStore using the follow sample replacing the necessary `<variables>`. 

```python
from semantic_kernel.connectors.ai.open_ai import AzureTextEmbedding

text_embedding = AzureTextEmbedding(
        service_id="default",
        endpoint="<my_endpoint>",
        api_key="<api_key>",
        deployment_name="<deployment_name>")

from azure.identity import DefaultAzureCredential
from semantic_kernel.connectors.memory.azure_cognitive_search import AzureCognitiveSearchMemoryStore

acs_memory_store = AzureCognitiveSearchMemoryStore(
        vector_size=1536,
        search_endpoint="<my_endpoint>", 
        token_credentials=DefaultAzureCredential())

memory = SemanticTextMemory(storage=acs_memory_store, embeddings_generator=text_embedding)
``` 


2. See error

```
File ~/miniconda3/envs/RVC3.12-env/lib/python3.12/site-packages/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py:83, in AzureCognitiveSearchMemoryStore.__init__(self, vector_size, search_endpoint, admin_key, azure_credentials, token_credentials, env_file_path)
     [76](https://file+.vscode-resource.vscode-cdn.net/Users/f/c/gh/prj/d/l/sk/notebooks/~/miniconda3/envs/p3.12/lib/python3.12/site-packages/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py:76)     logger.warning(f"Failed to load AzureAISearch pydantic settings: {e}")
     [78](https://file+.vscode-resource.vscode-cdn.net/Users/f/c/gh/prj/d/l/sk/notebooks/~/miniconda3/envs/p3.12/lib/python3.12/site-packages/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py:78) admin_key = admin_key or (
     [79](https://file+.vscode-resource.vscode-cdn.net/Users/f/c/gh/prj/d/l/sk/notebooks/~/miniconda3/envs/p3.12/lib/python3.12/site-packages/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py:79)     acs_memory_settings.api_key.get_secret_value()
     [80](https://file+.vscode-resource.vscode-cdn.net/Users/f/c/gh/prj/d/l/sk/notebooks/~/miniconda3/envs/p3.12/lib/python3.12/site-packages/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py:80)     if acs_memory_settings and acs_memory_settings.api_key
     [81](https://file+.vscode-resource.vscode-cdn.net/Users/f/c/gh/prj/d/l/sk/notebooks/~/miniconda3/envs/p3.12/lib/python3.12/site-packages/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py:81)     else None
     [82](https://file+.vscode-resource.vscode-cdn.net/Users/f/c/gh/prj/d/l/sk/notebooks/~/miniconda3/envs/p3.12/lib/python3.12/site-packages/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py:82) )
---> [83](https://file+.vscode-resource.vscode-cdn.net/Users/f/c/gh/prj/d/l/sk/notebooks/~/miniconda3/envs/p3.12/lib/python3.12/site-packages/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py:83) assert admin_key, "The ACS admin_key is required to connect to Azure Cognitive Search."
     [84](https://file+.vscode-resource.vscode-cdn.net/Users/f/c/gh/prj/d/l/sk/notebooks/~/miniconda3/envs/p3.12/lib/python3.12/site-packages/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py:84) search_endpoint = search_endpoint or (
     [85](https://file+.vscode-resource.vscode-cdn.net/Users/f/c/gh/prj/d/l/sk/notebooks/~/miniconda3/envs/p3.12/lib/python3.12/site-packages/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py:85)     acs_memory_settings.endpoint if acs_memory_settings and acs_memory_settings.endpoint else None
     [86](https://file+.vscode-resource.vscode-cdn.net/Users/f/c/gh/prj/d/l/sk/notebooks/~/miniconda3/envs/p3.12/lib/python3.12/site-packages/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py:86) )
     [87](https://file+.vscode-resource.vscode-cdn.net/Users/f/c/gh/prj/d/l/sk/notebooks/~/miniconda3/envs/p3.12/lib/python3.12/site-packages/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py:87) assert search_endpoint, "The ACS endpoint is required to connect to Azure Cognitive Search."

AssertionError: The ACS admin_key is required to connect to Azure Cognitive Search.
``` 

**Expected behavior**
Expect to be able to create the SearchIndexClient() in the utils.py file using the TokenCredential. 

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Platform**
 - OS: All
 - IDE: VS Code
 - Language: Python
 - Source: 1.0.0rc1

**Additional context**

--- a/python/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py
+++ b/python/semantic_kernel/connectors/memory/azure_cognitive_search/azure_cognitive_search_memory_store.py
@@ -81,7 +81,6 @@ class AzureCognitiveSearchMemoryStore(MemoryStoreBase):
             if acs_memory_settings and acs_memory_settings.api_key
             else None
         )
-        assert admin_key, "The ACS admin_key is required to connect to Azure Cognitive Search."
         search_endpoint = search_endpoint or (
             acs_memory_settings.endpoint if acs_memory_settings and acs_memory_settings.endpoint else None
         )

